### PR TITLE
test: screenshot on failure

### PR DESCRIPTION
**Pull request by JoelEinbinder**

### CODE REVIEW

File: test/jest/playwrightEnvironment.js [hunk 1]
```diff
@@ -114,15 +114,34 @@ class PlaywrightEnvironment extends NodeEnvironment {
...
-        }
+        return await this.fixturePool.resolveParametersAndRun(fn);
       };
     }
 
+    if (event.name === 'test_fn_success')
+      await this.fixturePool.teardownScope('test');
+
     if (event.name === 'test_fn_failure') {
+      let index = 0;
```
Comment (pavelfeldman) [inline test/jest/playwrightEnvironment.js:125]: What about timeouts?

**Review by pavelfeldman — APPROVED** [review]

File: test/jest/playwrightEnvironment.js (hunk 1 above)
Comment (JoelEinbinder) [reply test/jest/playwrightEnvironment.js:125]: Timeouts are failures are they not?

Comment (pavelfeldman) [reply test/jest/playwrightEnvironment.js:125]: They are and I don't think your solution covers them.

Comment (JoelEinbinder) [reply test/jest/playwrightEnvironment.js:125]: I tested it with timeouts and it correctly produces images.

**Comment by pgoos** [comment]
@pavelfeldman @JoelEinbinder Are there any guidelines how to enable screenshots on test failure?

**Comment by mxschmitt** [comment]
This functionality needs to be implemented on the test runner, Playwright itself is only a library to automate browsers. For example [Jest](https://github.com/playwright-community/jest-playwright#usage-with-custom-testenvironment) or [playwright-test](https://github.com/microsoft/playwright-test) (official test-runner but still beta, should be ready in a few weeks/months).

**Pull request merged**